A number of updates and issue fixes, updated to use v1.1 of happy app.

andyb3 opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/moka-guys/benchmarking_tool/16)
<!-- Reviewable:end -->

aledj2 commented:
Reviewed 4 of 7 files at r1.
Review status: 4 of 7 files reviewed at latest revision, 32 unresolved discussions.

---

*[happy_vcfeval/forms.py, line 17 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyadPYVFExaaTUubF5h:-KyadPYWuIzsU1JBHjWC:b-bcj8fs) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/forms.py#L17)):*
> ```Python
>         # Check uploaded vcf size does not exceed 200MB
>         if uploaded_vcf._size > 200000000:
>             raise forms.ValidationError({'vcf_file': "File size cannot be greater than 200MB"})
> ```

If this is displayed to the user can this message state that the vcf can't be greater than ...

---

*[happy_vcfeval/forms.py, line 20 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyadgD9bd24cq_TqZ4v:-KyadgD9bd24cq_TqZ4w:b-phvziw) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/forms.py#L20)):*
> ```Python
>         # Check uploaded file has .vcf or .vcf.gz extension
>         if not uploaded_vcf.name.endswith(".vcf.gz") and not uploaded_vcf.name.endswith(".vcf"):
>             raise forms.ValidationError({'vcf_file': "File must be VCF (.vcf) or gzipped VCF ('.vcf.gz')"})
> ```

again specify this refers to the vcf file

---

*[happy_vcfeval/forms.py, line 25 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyadlSrtEqpSEPq2u1a:-KyadlSsBNNzcxtPAVWe:bfwg68n) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/forms.py#L25)):*
> ```Python
>             # Check uploaded bed size does not exceed 200MB
>             if uploaded_bed._size > 200000000:
>                 raise forms.ValidationError({'bed_file': "File size cannot be greater than 200MB"})
> ```

state bed file can't be greater than 200MB

---

*[happy_vcfeval/precision_medicine.py, line 11 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyafcCylzxDnCUhznYS:-KyafcCylzxDnCUhznYT:b-8vsisx) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L11)):*
> ```Python
> from django.conf import settings
> from email.Message import Message
> import precision_medicine_config as config
> ```

specify this is the config file - some of the config file could be shared - discussed and it's backed up on mokanas

---

*[happy_vcfeval/precision_medicine.py, line 14 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyauVw1xPoQV0YdM8Z5:-KyauVw1xPoQV0YdM8Z6:b7jo6lq) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L14)):*
> ```Python
> 
> 
> class upload2Nexus(object):
> ```

how is this script called?

---

*[happy_vcfeval/precision_medicine.py, line 15 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyaeGyFCCMzulWXiuJ_:-KyaeGyFCCMzulWXiuJa:b-wlln7y) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L15)):*
> ```Python
> 
> class upload2Nexus(object):
>     ''' Submits jobs to dnanexus_happy app from web interface, and returns results via email '''
> ```

doc strings should be double quotes (sorry) - please do for all functions

---

*[happy_vcfeval/precision_medicine.py, line 79 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyahmTQBlHT93x6ofot:-KyahmTQBlHT93x6ofou:bq8vez9) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L79)):*
> ```Python
>         self.directory = os.path.dirname(self.vcf_filepath)
> 
>         if bed_file:
> ```

please comment (although the code does make it pretty obvious what is happening!)

---

*[happy_vcfeval/precision_medicine.py, line 82 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyai5Lt4SnOu8vBKdv0:-Kyai5LuJ0ekPHy9WuMQ:b-gmv3eo) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L82)):*
> ```Python
>             self.bed_filepath = bed_file
>             # build path and filename
>             self.bed_basename = os.path.basename(self.bed_filepath)
> ```

what does basename do?

---

*[happy_vcfeval/precision_medicine.py, line 85 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyai9fgMKT9UACOwv2d:-Kyai9fgMKT9UACOwv2e:b9rhuu4) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L85)):*
> ```Python
> 
>             # capture timestamp
>         self.timestamp = self.directory.split("/")[-1]
> ```

what does self.directory look like? Why are you splitting on '/' and taking the last element?

---

*[happy_vcfeval/precision_medicine.py, line 99 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyaihHtWt182EjA_Wgv:-KyaihHtWt182EjA_Wgw:b-301ttu) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L99)):*
> ```Python
>         self.vcf_strip()
> 
>     def vcf_strip(self):
> ```

doc string

---

*[happy_vcfeval/precision_medicine.py, line 108 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyajF6QEegpHRJOqeiD:-KyajF6QEegpHRJOqeiE:bprmbjm) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L108)):*
> ```Python
>         self.logfile.close()
>         # set query vcf
>         query_vcf = self.vcf_filepath
> ```

could you use self.vcf_filepath and self.vcfheader instead of these variables?

---

*[happy_vcfeval/precision_medicine.py, line 125 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyajylu-abfA28rx0_C:-KyajylvqAA9PYzhiWcz:b-a76tzg) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L125)):*
> ```Python
>             with open(vcf_header, 'r') as t, open_func(query_vcf, open_mode) as q, gzip.open(output_vcf, 'wb') as o:
>                 # for each line in q if it's not a header take the first 6 columns of each row, then add two full stops
>                 # (replacing the filter and info ), then just include the GT field of format and sample columns.
> ```

this does make sense but is quite hard to follow, could you add some more comments explaining the code please, mostly describing how you are accessing the GT fields from vcf 

---

*[happy_vcfeval/precision_medicine.py, line 149 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyakPwCpRCZ_4v9k9jV:-KyakPwCpRCZ_4v9k9jW:bnocslz) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L149)):*
> ```Python
>             self.email_message = ("An error was encountered whilst reading VCF:\n" + self.vcf_basename_orig
>                                   + "\n\nPlease ensure that the VCF (.vcf) or gzipped VCF (.vcf.gz) file supplied "
>                                     "conforms the VCF specification, is sorted, and includes genotype information "
> ```

conforms to 

---

*[happy_vcfeval/precision_medicine.py, line 161 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyalNhA2UU98RwE5EK_:-KyalNhA2UU98RwE5EKa:b-2gmq0v) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L161)):*
> ```Python
> 
>             # exit
>             sys.exit()
> ```

stop the script because vcf stripping failed

---

*[happy_vcfeval/precision_medicine.py, line 185 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyalpMHVs0uwyKbJIxW:-KyalpMHVs0uwyKbJIxX:b2si60m) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L185)):*
> ```Python
>         # eg path/to/ua  --auth-token abc  --project  projectname --folder /nexus/path --do-not-compress /file/to/upload
>         upload_cmd = (self.upload_agent + self.auth + self.nexusprojectstring + config.data_project_id.replace(":", "")
>                       + self.dest + self.nexus_folder + self.end_of_upload + "'{}'".format(self.vcf_filepath))
> ```

what does the format function do?

---

*[happy_vcfeval/precision_medicine.py, line 187 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyam-c6s-Y4NZ0xrh4Q:-Kyam-c6s-Y4NZ0xrh4R:bo2omff) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L187)):*
> ```Python
>                       + self.dest + self.nexus_folder + self.end_of_upload + "'{}'".format(self.vcf_filepath))
>         if self.bed_filepath:
>             upload_cmd += ("\n" + self.upload_agent + self.auth + self.nexusprojectstring
> ```

maybe test for bedfile first, if so concatenate two file paths and then do a single upload using the concatenated file paths? (although this works just as well!)

---

*[happy_vcfeval/precision_medicine.py, line 202 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyap8QRsnfgot9kukWG:-Kyap8QRsnfgot9kukWH:b-sgp42s) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L202)):*
> ```Python
>                                 , shell=True)
> 
>         # capture the streams (err is redirected to out above)
> ```

this comment is not accurate. what if there is any standard error? should it be written to log file at least? 

---

*[happy_vcfeval/precision_medicine.py, line 229 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyapwA9CGDzhqQg_DMu:-KyapwAAKbMaMRD8z7EV:b-wd3sty) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L229)):*
> ```Python
>         run_bash_script = open(run_bash_script_name, 'w')
> 
>         if self.bed_filepath:
> ```

what are you trying to do here?

---

*[happy_vcfeval/precision_medicine.py, line 233 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyaq5fqTDIu0nQ6cVCW:-Kyaq5fqTDIu0nQ6cVCX:b-prr1vg) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L233)):*
> ```Python
>                 config.data_project_id + self.nexus_folder + "/" + self.bed_basename)
>         else:
>             self.app_panel_bed = config.app_panel_bed
> ```

what is config.app_panel_bed?

---

*[happy_vcfeval/precision_medicine.py, line 235 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyaqOH0Qp5ETnRQSXwe:-KyaqOH0Qp5ETnRQSXwf:b-ocyjpi) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L235)):*
> ```Python
>             self.app_panel_bed = config.app_panel_bed
>         # dx run  command
>         # eg dxrun_cmd=self.base_cmd+workflow_query_vcf + project+self.nexus_folder +"/"+ self.vcf_basename
> ```

can you update the command you are trying to build please

---

*[happy_vcfeval/precision_medicine.py, line 253 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyaqead9NMmmnyr3zos:-Kyaqead9NMmmnyr3zot:b-9cmyad) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L253)):*
> ```Python
>         run_bash_script.close()
> 
>         # run the command
> ```

run bash script containing the dx run commands

---

*[happy_vcfeval/precision_medicine.py, line 260 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyar38IHEUv-sO36h-D:-Kyar38JnAgbcA6i9Mmu:bryizoc) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L260)):*
> ```Python
>         (out, err) = proc.communicate()
>         if err:
>             # send a error email
> ```

i guess this is sending an email to moka guys?

---

*[happy_vcfeval/precision_medicine.py, line 268 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyar8TsEECdXRu5ddV1:-Kyar8TsEECdXRu5ddV2:bh4se9g) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L268)):*
> ```Python
> 
>             # send a error email to user
>             self.you = [self.email]
> ```

overwrite the email address and subject etc in config file once it's sent an email to mokaguys so it now sends an email to the user?

---

*[happy_vcfeval/precision_medicine.py, line 327 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyarwD42lU9yQSyIe8K:-KyarwD42lU9yQSyIe8L:b-pytdng) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L327)):*
> ```Python
>         fail_status_bash_script.close()
> 
>         count = 0
> ```

comment (even though it's pretty obvious)

---

*[happy_vcfeval/precision_medicine.py, line 345 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyas_daXPT_TAqRHqEi:-Kyas_daXPT_TAqRHqEj:b87j5df) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L345)):*
> ```Python
>                 time.sleep(60)
>                 # call the check status function again
>                 self.check_status(status_bash_script_name)
> ```

is this step needed? after waiting for a minute the loop will restart by checking the status so by checking here you check twice in quick succession, and if true it wouldn't exit the loop.

---

*[happy_vcfeval/precision_medicine.py, line 354 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyasxvgYB_9Pb4knkEG:-KyasxvgYB_9Pb4knkEH:b3q9huh) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L354)):*
> ```Python
>                 # if has been running for 45 mins stop or the job has failed
>                 # access the error message from the app:
>                 # command which returns a job-id within the project if successfully completed
> ```

is this comment accurate?

---

*[happy_vcfeval/precision_medicine.py, line 461 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyatZ8tBFqV-0JUZD-N:-KyatZ8tBFqV-0JUZD-O:bzh0py3) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L461)):*
> ```Python
>         # if error
>         if err:
>             self.logfile = open(self.logfile_name, 'a')
> ```

it's almost worth making this a function which takes whatever it needs to write as an argument and call it each time! (open logfile (can opening it as append create a file?), write then close)

---

*[happy_vcfeval/precision_medicine.py, line 485 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyau-6tQhxVb1pmBDMT:-Kyau-6tQhxVb1pmBDMU:bqf8hqa) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L485)):*
> ```Python
> 
>             # open the extended summary file to get recall and precision
>             summary_csv = (tarfile.open(os.path.join(self.directory, "happy." + self.vcf_basename_orig.split(".vcf")[0]
> ```

what are you doing here?

---

*[happy_vcfeval/views.py, line 15 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyaubKf-0Ug1VweL-OY:-KyaubKf-0Ug1VweL-OZ:b-jrxcue) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/views.py#L15)):*
> ```Python
> # Upload to nexus and run app
> def run_prec_med(email, vcf_filepath, bed_filepath):
>     upload = precision_medicine.upload2Nexus()
> ```

comment

---

*[happy_vcfeval/views.py, line 55 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyavBLDThvJsEbWAjeO:-KyavBLDThvJsEbWAjeP:b93da0w) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/views.py#L55)):*
> ```Python
>                 bed_filepath = settings.MEDIA_ROOT + timestamp + "/" + bed_filename
>                 os.rename(bed_filepath_orig, bed_filepath)
>                 # Run dna_nexus in background thread. Calls run_prec_med() function.
> ```

pass file paths to function (including bedfile)

---

*[happy_vcfeval/views.py, line 66 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyavL4EXxXUTkqUBCZP:-KyavL4EXxXUTkqUBCZQ:bh26qns) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/views.py#L66)):*
> ```Python
>             t.start()  # Start the thread, which calls run_prec_med() function
>             # Redirect to the 'processing' success page.
>             return redirect(reverse('happy_vcfeval:processing'))
> ```

what does reverse mean/do?

---

*[happy_vcfeval/views.py, line 68 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyava4jnMAsZSLMFM4Y:-Kyava4jnMAsZSLMFM4Z:bkj6pfh) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/views.py#L68)):*
> ```Python
>             return redirect(reverse('happy_vcfeval:processing'))
>         else:
>             # If validation failed, reload the form which will display the error messages from failed validation
> ```

where does the error message get stated/edf into form?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16)*
<!-- Sent from Reviewable.io -->

andyb3 commented:
Review status: 4 of 7 files reviewed at latest revision, 30 unresolved discussions.

---

*[happy_vcfeval/precision_medicine.py, line 11 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyafcCylzxDnCUhznYS:-KyvbCa-6a2prure1EhP:b3ka5zh) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L11)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

specify this is the config file - some of the config file could be shared - discussed and it's backed up on mokanas
</blockquote></details>

Updated comment and raised issue to separate config into public and private in next release

---

*[happy_vcfeval/precision_medicine.py, line 14 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyauVw1xPoQV0YdM8Z5:-KyvbMDpNc0Z4hJMm_pX:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L14)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

how is this script called?
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 15 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyaeGyFCCMzulWXiuJ_:-KyvbNAddA1GBCAoPKup:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L15)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

doc strings should be double quotes (sorry) - please do for all functions
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 79 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyahmTQBlHT93x6ofot:-KyvbQepy_6-toFaGTYs:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L79)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

please comment (although the code does make it pretty obvious what is happening!)
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 82 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyai5Lt4SnOu8vBKdv0:-KyvbTIy07FwK7E_jqq-:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L82)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

what does basename do?
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 85 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyai9fgMKT9UACOwv2d:-KyvbVk73PIpNSOsF6hg:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L85)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

what does self.directory look like? Why are you splitting on '/' and taking the last element?
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 99 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyaihHtWt182EjA_Wgv:-KyvbYDh-XV7e0kXu8W7:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L99)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

doc string
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 125 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyajylu-abfA28rx0_C:-KyvbbadebqZs4TGd-0C:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L125)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

this does make sense but is quite hard to follow, could you add some more comments explaining the code please, mostly describing how you are accessing the GT fields from vcf 
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 149 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyakPwCpRCZ_4v9k9jV:-Kyvbdeki5SUm76PrPsk:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L149)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

conforms to 
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 161 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyalNhA2UU98RwE5EK_:-KyvbfSTKB3eLEJY1Ntm:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L161)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

stop the script because vcf stripping failed
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 185 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyalpMHVs0uwyKbJIxW:-KyvbiYExzNMR0D7xVIX:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L185)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

what does the format function do?
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 229 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyapwA9CGDzhqQg_DMu:-KyvbnG_80QuBL1Vzfpr:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L229)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

what are you trying to do here?
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 233 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyaq5fqTDIu0nQ6cVCW:-KyvbpOpVgAp0CRj7Paz:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L233)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

what is config.app_panel_bed?
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 235 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyaqOH0Qp5ETnRQSXwe:-Kyvc-lc3OU6TiKPSKfl:bfkkszb) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L235)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

can you update the command you are trying to build please
</blockquote></details>

Changed comment to make it more descriptive

---

*[happy_vcfeval/precision_medicine.py, line 253 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyaqead9NMmmnyr3zos:-Kyvc58ppb7G-Tnvzghp:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L253)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

run bash script containing the dx run commands
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 260 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyar38IHEUv-sO36h-D:-Kyvc6pNz935n0I40cD6:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L260)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

i guess this is sending an email to moka guys?
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 268 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyar8TsEECdXRu5ddV1:-Kyvc8pOZZa_j7wcamIX:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L268)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

overwrite the email address and subject etc in config file once it's sent an email to mokaguys so it now sends an email to the user?
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 327 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyarwD42lU9yQSyIe8K:-KyvcAxbpqRIOQOpq6tn:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L327)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

comment (even though it's pretty obvious)
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 354 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyasxvgYB_9Pb4knkEG:-KyvcDi0JPGjD-QWF3zM:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L354)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

is this comment accurate?
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 485 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyau-6tQhxVb1pmBDMT:-KyvcGjerrF-9IxjsSIJ:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L485)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

what are you doing here?
</blockquote></details>

Done.

---

*[happy_vcfeval/views.py, line 15 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyaubKf-0Ug1VweL-OY:-KyvcJ6gwaJqWDOo_qI8:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/views.py#L15)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

comment
</blockquote></details>

Done.

---

*[happy_vcfeval/views.py, line 55 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyavBLDThvJsEbWAjeO:-KyvcK9wfGPmPojb4ZL9:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/views.py#L55)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

pass file paths to function (including bedfile)
</blockquote></details>

Done.

---

*[happy_vcfeval/views.py, line 66 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyavL4EXxXUTkqUBCZP:-KyvcM_1H-eHuOYxWXOC:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/views.py#L66)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

what does reverse mean/do?
</blockquote></details>

Done.

---

*[happy_vcfeval/views.py, line 68 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyava4jnMAsZSLMFM4Y:-KyvcNqcW3Rp1aL2EI9g:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/views.py#L68)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

where does the error message get stated/edf into form?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16)*
<!-- Sent from Reviewable.io -->

andyb3 commented:
Review status: 4 of 7 files reviewed at latest revision, 28 unresolved discussions.

---

*[happy_vcfeval/forms.py, line 17 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyadPYVFExaaTUubF5h:-KyyQaGwyz5ZnjmBm6IR:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/forms.py#L17)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

If this is displayed to the user can this message state that the vcf can't be greater than ...
</blockquote></details>

Done.

---

*[happy_vcfeval/forms.py, line 20 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyadgD9bd24cq_TqZ4v:-KyyQbteSd78mQ0fwIIK:b-sklqdc) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/forms.py#L20)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

again specify this refers to the vcf file
</blockquote></details>

I haven't done this because I think it would be strange to write "VCF must be VCF", and the error messages are displayed under the input box they refer to (i.e. error messages relating to VCF file are displayed under the VCF upload field and BED messages dispayed under BED upload field). I've added a comment further up to clarify this.

---

*[happy_vcfeval/forms.py, line 25 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyadlSrtEqpSEPq2u1a:-KyyR_iBf0-cSCYq7FNg:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/forms.py#L25)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

state bed file can't be greater than 200MB
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16)*
<!-- Sent from Reviewable.io -->

andyb3 commented:
Review status: 4 of 7 files reviewed at latest revision, 30 unresolved discussions.

---

*[happy_vcfeval/precision_medicine.py, line 108 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyajF6QEegpHRJOqeiD:-KyyWq2BxNDd_LHF4Fcr:b-896fix) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L108)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

could you use self.vcf_filepath and self.vcfheader instead of these variables?
</blockquote></details>

Done.

---

*[happy_vcfeval/precision_medicine.py, line 187 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyam-c6s-Y4NZ0xrh4Q:-KyyUtXvimo3jFLlUvp2:b-6xldgw) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L187)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

maybe test for bedfile first, if so concatenate two file paths and then do a single upload using the concatenated file paths? (although this works just as well!)
</blockquote></details>

Agreed that would be better. I've raised an issue so it can be dealt with in the next release if that's OK?

---

*[happy_vcfeval/precision_medicine.py, line 345 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyas_daXPT_TAqRHqEi:-KyyWFZYgxORrwNY1_uQ:basxbbq) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L345)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

is this step needed? after waiting for a minute the loop will restart by checking the status so by checking here you check twice in quick succession, and if true it wouldn't exit the loop.
</blockquote></details>

You're right, it's gone.

---

*[happy_vcfeval/precision_medicine.py, line 461 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-KyatZ8tBFqV-0JUZD-N:-KyyWX3K28BmU5zqQU9j:b22paoo) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L461)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

it's almost worth making this a function which takes whatever it needs to write as an argument and call it each time! (open logfile (can opening it as append create a file?), write then close)
</blockquote></details>

Have raised this an issue to be dealt with in next release

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16)*
<!-- Sent from Reviewable.io -->

andyb3 commented:
Review status: 4 of 7 files reviewed at latest revision, 28 unresolved discussions.

---

*[happy_vcfeval/precision_medicine.py, line 202 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyap8QRsnfgot9kukWG:-KyyZXjevo6J3r8lpzzV:b-vfblvf) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L202)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

this comment is not accurate. what if there is any standard error? should it be written to log file at least? 
</blockquote></details>

Updated so stderr is redirected to stdout so both are captured in log file 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16)*
<!-- Sent from Reviewable.io -->

aledj2 commented:
Review status: 4 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

*[happy_vcfeval/precision_medicine.py, line 85 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyai9fgMKT9UACOwv2d:-Kz-TlEz9zvH_E1rvVPR:bj8q0oq) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L85)):*
<details><summary><i>Previously, andyb3 wrote…</i></summary><blockquote>

Done.
</blockquote></details>


sorry, could we have an example of self.directory eg an example file path

---

*[happy_vcfeval/precision_medicine.py, line 17 at r2](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kz-TQKOoIG6IFS7J8As:-Kz-TQKOoIG6IFS7J8At:b-ct4wf) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/6b72d8b76622773543335cabdc57bdb9baca6a68/happy_vcfeval/precision_medicine.py#L17)):*
> ```Python
>     """
>     Submits jobs to dnanexus_happy app from web interface, and returns results via email.
>     Takes an email address, vcf file and optional bed file as inputs (via take_inputs() method)
> ```

So the inputs are full file paths?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16)*
<!-- Sent from Reviewable.io -->

aledj2 commented:
Reviewed 2 of 3 files at r2.
Review status: 6 of 7 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16)*
<!-- Sent from Reviewable.io -->

andyb3 commented:
Review status: 6 of 7 files reviewed at latest revision, 2 unresolved discussions.

---

*[happy_vcfeval/precision_medicine.py, line 85 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyai9fgMKT9UACOwv2d:-Kz-ihCg_K_ufqdXN-Y1:b-tovr0p) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L85)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

sorry, could we have an example of self.directory eg an example file path
</blockquote></details>

Yes sorry should have put that in, I've added filepath and made the comment a bit clearer.

---

*[happy_vcfeval/precision_medicine.py, line 17 at r2](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kz-TQKOoIG6IFS7J8As:-Kz-i1VUaSmtp7nfwOw6:bkvbyvj) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/6b72d8b76622773543335cabdc57bdb9baca6a68/happy_vcfeval/precision_medicine.py#L17)):*
<details><summary><i>Previously, aledj2 (Aled Jones) wrote…</i></summary><blockquote>

So the inputs are full file paths?
</blockquote></details>

Sorry I should have said filepath rather than files. Also added an extra line that seems to have been shoved below the big grey box so don't miss it! Hopefully it's a bit clearer now

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16)*
<!-- Sent from Reviewable.io -->

aledj2 commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[happy_vcfeval/precision_medicine.py, line 85 at r1](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kyai9fgMKT9UACOwv2d:-Kz-jVqtM6QcvcMxETF4:b-k73z76) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/ffaa29f7ce06dde196e0d77ebabe42de871e7854/happy_vcfeval/precision_medicine.py#L85)):*
<details><summary><i>Previously, andyb3 wrote…</i></summary><blockquote>

Yes sorry should have put that in, I've added filepath and made the comment a bit clearer.
</blockquote></details>

thanks

---

*[happy_vcfeval/precision_medicine.py, line 17 at r2](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16#-Kz-TQKOoIG6IFS7J8As:-Kz-jH1cSUE7T7JHPEkf:b-k73z76) ([raw file](https://github.com/moka-guys/benchmarking_tool/blob/6b72d8b76622773543335cabdc57bdb9baca6a68/happy_vcfeval/precision_medicine.py#L17)):*
<details><summary><i>Previously, andyb3 wrote…</i></summary><blockquote>

Sorry I should have said filepath rather than files. Also added an extra line that seems to have been shoved below the big grey box so don't miss it! Hopefully it's a bit clearer now
</blockquote></details>

thanks

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/moka-guys/benchmarking_tool/16)*
<!-- Sent from Reviewable.io -->
